### PR TITLE
Feature: Add endpoint for creating a ticket type for an event

### PR DESCRIPTION
**Pull request by kjohnson**
Resolves [GIVE-365](https://stellarwp.atlassian.net/browse/GIVE-365)

## Description

This PR adds a new WP Rest Route for creating a new ticket type for an event as a POST request.

Note: This also resolves a small bug with the `created_at` and `updated_at` timestamps when creating a Ticket Type.

## Visuals

image

## Testing Instructions

Make a POST request to `/wp-json/give-api/v2/events-tickets/event/{event_id}/ticket-types` passing the `label` (string), `description` (string), `price` (int), and `capacity` (int) and confirm a `201 Created` response.

[GIVE-365]: https://stellarwp.atlassian.net/browse/GIVE-365?atlOrigin=eyJpIjoiNWRkNTljNzYxNjVmNDY3MDlhMDU5Y2ZhYzA5YTRkZjUiLCJwIjoiZ2l0aHViLWNvbS1KU1cifQ

### CODE REVIEW

File: src/EventTickets/Routes/CreateEventTicketType.php [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+     *
+     */
+    public function handleRequest(WP_REST_Request $request)
+    {
+        $ticketType = EventTicketType::create([
+            'eventId' => $request->get_param('event_id'),
+            'title' => $request->get_param('title'),
+            'description' => $request->get_param('description'),
+            'price' => new Money($request->get_param('price'), give_get_currency()),
+            'maxAvailable' => $request->get_param('capacity'),
```
Comment (pauloiankoski) [inline src/EventTickets/Routes/CreateEventTicketType.php:86]: This param has been updated. It's `capacity` now.

File: src/EventTickets/Routes/CreateEventTicketType.php [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+        ]);
+
+        return new WP_REST_Response($ticketType->toArray(), 201);
```
Comment (pauloiankoski) [inline src/EventTickets/Routes/CreateEventTicketType.php:89]: In the future, if we choose to use this response, we will need to manually convert the `price` attribute to an array format.

Comment (kjohnson) [reply src/EventTickets/Routes/CreateEventTicketType.php:89]: I can replace the return with a DTO, which is probably better in light of some attributes being objects.

File: src/EventTickets/Routes/CreateEventTicketType.php (hunk 1 above)
Comment (kjohnson) [reply src/EventTickets/Routes/CreateEventTicketType.php]: Resolved, see [fee2beb](https://github.com/impress-org/givewp/pull/7252/commits/fee2beb85ff45be994f1af73ae8f2250e844fd37).

File: src/EventTickets/Routes/CreateEventTicketType.php (hunk 2 above)
Comment (kjohnson) [reply src/EventTickets/Routes/CreateEventTicketType.php]: Resolved, see [dc51f0b](https://github.com/impress-org/givewp/pull/7252/commits/dc51f0b65c38ad08b0bdaa3caf7a231277b3b833).

**Review by pauloiankoski — APPROVED** [review]

**Pull request merged**